FileProviderExtension/FileProviderExtension.swift

```diff
@@ -259,8 +259,9 @@ class FileProviderExtension: NSFileProviderExtension {
 		                                                              dbPath: dbPath,
 		                                                              delegate: LocalURLProvider(domain: snapshotDomain)))
 		#else
-		if let domain = domain, let localURLProvider = localURLProvider {
+		if let domain = domain, let localURLProvider = localURLProvider, let dbPath = dbPath, let notificator = notificator {
 			serviceSources.append(VaultUnlockingServiceSource(domain: domain, notificator: notificator, dbPath: dbPath, delegate: localURLProvider))
+			serviceSources.append(UploadRetryingServiceSource(domain: domain, notificator: notificator, dbPath: dbPath, delegate: localURLProvider))
```

Comment:
Seems like a slightly different pattern now for `VaultUnlockingServiceSource`. It's a small thing but would it make sense to refactor `VaultUnlockingServiceSource` to accept a non-optional `notificator` as well?

Reply:
Indeed, however I wanted to refactor this after the PR, as I don't think it really has anything to do with the functionality that was introduced in this PR.